src/commands/lambda/constants.ts

```diff
@@ -1,34 +1,50 @@
 export const DD_LAMBDA_EXTENSION_LAYER_NAME = 'Datadog-Extension'
 export const EXTENSION_LAYER_KEY = 'extension'
+export const DOTNET_RUNTIME = 'dotnetcore3.1'
 export const RUNTIME_LAYER_LOOKUP = {
   [EXTENSION_LAYER_KEY]: DD_LAMBDA_EXTENSION_LAYER_NAME,
+  'dotnetcore3.1': 'poc-maxday-dotnet',
+  'java11': DD_LAMBDA_EXTENSION_LAYER_NAME,
+  'java8.al2': DD_LAMBDA_EXTENSION_LAYER_NAME,
   'nodejs12.x': 'Datadog-Node12-x',
   'nodejs14.x': 'Datadog-Node14-x',
+  'provided.al2': DD_LAMBDA_EXTENSION_LAYER_NAME,
   'python3.6': 'Datadog-Python36',
   'python3.7': 'Datadog-Python37',
   'python3.8': 'Datadog-Python38',
   'python3.9': 'Datadog-Python39',
+  'ruby2.5': 'Datadog-Ruby2-5',
+  'ruby2.7': 'Datadog-Ruby2-7',
 } as const
 // We exclude the Extension Layer Key in order for the runtime
 // to be used directly in HANDLER_LOCATION.
 export type Runtime = Exclude<keyof typeof RUNTIME_LAYER_LOOKUP, typeof EXTENSION_LAYER_KEY>
-
-export const ARM_RUNTIMES = [EXTENSION_LAYER_KEY, 'python3.8', 'python3.9']
+export const ARM_RUNTIMES = [EXTENSION_LAYER_KEY, 'java8.al2', 'java11', 'provided.al2', 'python3.8', 'python3.9']
 export const ARM64_ARCHITECTURE = 'arm64'
 export const ARM_LAYER_SUFFIX = '-ARM'
 
 export enum RuntimeType {
+  DOTNET,
+  GO,
+  JAVA,
   NODE,
   PYTHON,
+  RUBY,
 }
 
-export const RUNTIME_LOOKUP: {[key: string]: RuntimeType} = {
+export const RUNTIME_LOOKUP: { [key: string]: RuntimeType } = {
+  'dotnetcore3.1': RuntimeType.DOTNET,
+  'java11': RuntimeType.JAVA,
+  'java8.al2': RuntimeType.JAVA,
+  'provided.al2': RuntimeType.GO,
```

Comment:
Technically `provided.al2` doesn't have to be Go. It's a custom runtime. It could be Go, but it could also be anything else.